File: sdk/node/examples/create-session.ts
import { DigiPay } from '@dgbwallet/digipay';

// Run with:  DIGIPAY_KEY=dgp_… npx tsx examples/create-session.ts

const dp = new DigiPay({ apiKey: process.env.DIGIPAY_KEY! });

const session = await dp.sessions.create({
    amount: 5,
    label: 'Order #1234',
    memo: 'Customer: alice@example.com',
});

console.log('Session ID:    ', session.id);
console.log('Amount:        ', session.amount, 'DGB');
console.log('Address:       ', session.address);
console.log('Expires:       ', session.expiresAt);
console.log('BIP21 URI:     ', session.uri);
console.log('Hosted page:   ', session.checkoutUrl);

File: sdk/python/examples/create_session.py
"""Create a checkout session.

Run with:
    DIGIPAY_KEY=dgp_… python examples/create_session.py
"""

from __future__ import annotations

import os

from digipay import DigiPay

dp = DigiPay(api_key=os.environ["DIGIPAY_KEY"])

session = dp.sessions.create(
    amount=5,
    label="Order #1234",
    memo="Customer: alice@example.com",
)

print("Session ID:  ", session["id"])
print("Amount:      ", session["amount"], "DGB")
print("Address:     ", session["address"])
print("Expires:     ", session["expiresAt"])
print("BIP21 URI:   ", session["uri"])
print("Hosted page: ", session["checkoutUrl"])
